src/components/CloudinaryInput.tsx

```diff
@@ -17,24 +17,64 @@ const CloudinaryInput = (props: ObjectInputProps) => {
   const handleSelect = useCallback(
     (payload: InsertHandlerParams) => {
       const [asset] = payload.assets
+
       if (!asset) {
         return
       }
 
-      onChange(
-        PatchEvent.from([
-          set(
-            Object.assign(
-              {
-                _type: type.name,
-                _version: 1,
-                ...(value?._key ? {_key: value._key} : {_key: nanoid()}),
-              },
-              asset
-            )
-          ),
-        ])
-      )
+      //The cloudinary metadata (context) needs to be transformed to a valid object key
+      //because the key can't contain special characters in the Studio
+      if (asset.context) {
+        const customKey = Object.keys(asset.context.custom)[0]
+        const customValue = asset.context.custom[customKey]
+        const customRegex = customKey.replace(/[^a-zA-Z0-9_]|-/g, '_')
```

Comment:
naming wise I think it would be clearer it this was named `renamedField`, as `customRegex` implies it points at a regex value